ghmirror/decorators/checks.py

```diff
@@ -0,0 +1,69 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Amador Pahim <apahim@redhat.com>
+
+"""
+Contains all the required verification
+"""
+
+import os
+
+from functools import wraps
+
+import flask
+
+from ghmirror.core.constants import GH_API
+from ghmirror.core.mirror_requests import conditional_request
+
+
+AUTHORIZED_USERS = os.environ.get('GITHUB_USERS')
+
+
+def check_user(function):
+    """
+    Checks whether the user is a member of one of the
+    authorized organizations
+    """
+    @wraps(function)
+    def wrapper(*args, **kwargs):
+        # When the GITHUB_USERS is not set, we resume the normal
+        # operation, where we do not check users
+        if AUTHORIZED_USERS is None:
+            return function(*args, **kwargs)
+
+        authorized_users = AUTHORIZED_USERS.split(':')
+        authorization = flask.request.headers.get('Authorization')
+        # At this stage, Authorization header is mandatory
+        if authorization is None:
+            return flask.Response('Authorization token required', 401)
+
+        # Using the Authorization header to get the user information
+        user_url = f'{GH_API}/user'
+        resp = conditional_request(method='GET', url=user_url,
+                                   auth=authorization)
+
+        # Returning whatever it is, if it is not a success
+        if resp.status_code != 200:
+            return flask.Response(resp.content, resp.status_code)
+
+        user_info = resp.json()
+        # At this point we have the authorized_users list and the
+        # user information from Github. If there's a match, we just
+        # return the decorated function
+        if user_info['login'] in authorized_users:
+            return function(*args, **kwargs)
+
+        # No match means user is not authorized
+        return flask.Response('User is an authorized user', 401)
```

Comment:
nit: The message may be misleading. Shouldn't it say something like User not being authorized as the comment suggests :)

Reply:
Fixed!